### PR TITLE
Handle allowed messages before initialize

### LINKED ISSUE #654: Handle allowed messages from server while initializing.
As would have been helpful in the #613 troubleshooting:

> during the initialize request the server is allowed to send the notifications 
> window/showMessage, window/logMessage and telemetry/event as well as the window/showMessageRequest request to the client.

### PR DESCRIPTION
**Pull request by rwols**
I had to move the MockWindow and MockView to test_sublime.py for this.

close #654

### CODE REVIEW

**Comment by rwols** [comment]
> Make the caller explicitly call initialize?

Tried this first, got endlessly stuck on tests.

> expose an on_initializing parameter in the Session constructor

Taken this route now. It works!

File: plugin/core/test_session.py [hunk 1]
```diff
@@ -89,39 +89,62 @@ def test_can_create_session(self):
 
     def test_can_get_started_session(self):
         project_path = "/"
-        created_callback = unittest.mock.Mock()
+        post_initialize_callbacak = unittest.mock.Mock()
```
Comment (tomv564) [inline plugin/core/test_session.py]: small typo: "callbacak"

Comment (rwols) [reply plugin/core/test_session.py]: Hmm, does this test not run then?

Comment (rwols) [reply plugin/core/test_session.py]: Never mind, I mindlessly used CTRL+D to change that variable name.

**Pull request merged**